GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -596,14 +599,15 @@ private void RefreshRevisions(Func<RefsFilter, IReadOnlyList<IGitRef>> getRefs)
                 return;
             }
 
+            InternalInitialize();
```

Comment:
This will delay the UI refresh slightly, the Git commands for revisions should be started async early

Reply:
It solved the issue for one of two cases. But `DoEvents` suffices. Reverted this part.